### PR TITLE
add group_by_os_family in azure dynamic inventory

### PR DESCRIPTION
**Pull request by EtienneDeneuve**
##### SUMMARY

Add a way to group by os family. 

I need that for set correct winrm/ssh connection in group_vars

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

azure_rm.py
azure_rm.ini

##### ANSIBLE VERSION

```
ansible 2.5.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/neocaseadmin/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Review by yuwzho — APPROVED** [review]

**Comment by EtienneDeneuve** [comment]
Where is the doc of azure_rm inventory ? I need to update it to add what provide my few additions to initial script.

**Comment by yuwzho** [comment]
@EtienneDeneuve Doc is here: https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/scenario_guides/guide_azure.rst

**Review by yungezz — APPROVED** [review]

File: contrib/inventory/azure_rm.py [hunk 1]
```diff
@@ -808,10 +810,16 @@ def _add_host(self, vars):
 
         host_name = self._to_safe(vars['name'])
         resource_group = self._to_safe(vars['resource_group'])
+        operating_system_type = self._to_safe(vars['os_disk']['operating_system_type'])
```
Comment (yungezz) [inline contrib/inventory/azure_rm.py]: the change looks good. pls update doc as @yuwzho  mentioned.

**Comment by EtienneDeneuve** [comment]
@yungezz 

The doc is updated.  

I think there's maybe an little enhancement, for me it's better to ensure the operating_system_type is in lower case, no? I'm not a python guru but I think we can change in 2 places:
```python
host_vars['os_disk'] = dict(
                name=machine.storage_profile.os_disk.name,
                operating_system_type=machine.storage_profile.os_disk.os_type.value
            )
```
by 
```python
host_vars['os_disk'] = dict(
                name=machine.storage_profile.os_disk.name,
                operating_system_type=machine.storage_profile.os_disk.os_type.value.tolower()
            )
```

Or in the group :
```Python
 def _add_host(self, vars):
        host_name = self._to_safe(vars['name'])
        resource_group = self._to_safe(vars['resource_group'])
        operating_system_type = self._to_safe(vars['os_disk']['operating_system_type'])
        security_group = None
```
by 

```Python
 def _add_host(self, vars):
        host_name = self._to_safe(vars['name'])
        resource_group = self._to_safe(vars['resource_group'])
        operating_system_type = self._to_safe(vars['os_disk']['operating_system_type'].tolower())
        security_group = None
```

I don't know which is the best/correct way to handle that. Azure send the data with Windows or Linux, and so in group_vars, we need to create Windows.yml or Linux.yml instead of lower case.
For me, the second option is better as we don't alter the Azure output in the inventory but create the dynamic group in lower case. I wait for your advice and update the PR with the correct way.

**Comment by yungezz** [comment]
@EtienneDeneuve each way is ok for me. I personal prefer to lower case  in _add_host.

**Comment by EtienneDeneuve** [comment]
I think it’s the cleaner way too. I’ll change it today

Obtenez Outlook pour iOS<https://aka.ms/o0ukef>
________________________________

File: contrib/inventory/azure_rm.py [hunk 2]
```diff
@@ -808,10 +810,16 @@ def _add_host(self, vars):
 
         host_name = self._to_safe(vars['name'])
         resource_group = self._to_safe(vars['resource_group'])
+        operating_system_type = self._to_safe(vars['os_disk']['operating_system_type'].tolower())
```
Comment (yungezz) [inline contrib/inventory/azure_rm.py]: lower()

**Comment by yungezz** [comment]
@jborean93 Hi Jordan, this PR adds a new option `group_by_os_family` to azure inventory. pls help to review when available. I've tested it locally.

File: docs/docsite/rst/scenario_guides/guide_azure.rst [hunk 3]
```diff
@@ -402,6 +404,12 @@ Here are some examples using the inventory script:
     # Execute /bin/uname on all instances in the Testing resource group
     $ ansible -i azure_rm.py Testing -m shell -a "/bin/uname -a"
 
+    # Execute win_ping on all Windows instances
+    $ ansible -i azure_rm.py Windows -m win_ping
```
Comment (yungezz) [inline docs/docsite/rst/scenario_guides/guide_azure.rst]: update sample to use lower case  os name

**Comment by yungezz** [comment]
@EtienneDeneuve will you update the PR per comments?  I can help after you give me repo permission.

**Comment by EtienneDeneuve** [comment]
Oh too bad I didn’t saw your comment until yet!

I’ll fix as mentioned today!

**Comment by EtienneDeneuve** [comment]
Normally fixed the 'tolower()' by 'lower()' and change in doc

**Comment by yungezz** [comment]
shipit

File: docs/docsite/rst/scenario_guides/guide_azure.rst [hunk 4]
```diff
@@ -402,6 +404,12 @@ Here are some examples using the inventory script:
     # Execute /bin/uname on all instances in the Testing resource group
     $ ansible -i azure_rm.py Testing -m shell -a "/bin/uname -a"
 
+    # Execute win_ping on all Windows instances
+    $ ansible -i azure_rm.py windows -m win_ping
+
+    # Execute win_ping on all Windows instances
+    $ ansible -i azure_rm.py winux -m ping
```
Comment (yungezz) [inline docs/docsite/rst/scenario_guides/guide_azure.rst:411]: typo: linux

**Pull request merged**